src/CloudFlare/Api.php

```diff
@@ -47,24 +55,45 @@ class Api
     private $permissions = null;
 
     /**
+     * Default API endpoint
+     */
+    protected $endpoint = self::ENDPOINT_DEFAULT;
+
+    /**
      * Make a new instance of the API client
      * This can be done via providing the email address and api key as seperate parameters
      * or by passing in an already instantiated object from which the details will be extracted
+     *
+     * Signature 1:
+     * @param $arg1 Object client object
+     *
+     * Signature 2:
+     * @param $arg1 string hosting partner key
+     *
+     * Signature 3:
+     * @param $arg1 string user email
+     * @param $arg2 string user auth key
      */
-    public function __construct()
+    public function __construct($arg1, $arg2 =  null, $arg3 = null)
```

Comment:
Hi

Just holding off on this now as CloudFlare have added a third parameter to their authentication methods: User Service Key (https://api.cloudflare.com/#endpoints). So I need to think about how this factors into this work.

Thanks,
James